spec/requests/teachers_spec.rb

```diff
@@ -0,0 +1,151 @@
+RSpec.describe Teacher, type: :request do
+  describe 'GET /index' do
+    let!(:teacher) { create(:teacher) }
+
+    it 'returns correct http status' do
+      get teachers_url
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'renders correct template' do
+      get '/teachers'
+
+      expect(response).to render_template(:index)
+    end
+  end
```

Comment:
Complementando o comentário do @mateuscelio, idealmente as mensagens podem ser mais assertivas para melhorar a clareza do que está sendo esperado no teste.
```suggestion
    it 'returns http status 200' do
      get teachers_url

      expect(response).to have_http_status(:ok)
    end

    it 'renders the index template' do
      get '/teachers'

      expect(response).to render_template(:index)
    end
  end
```

Reply:
Mesmo que o comentário abaixo: melhor escolher um entre `teachers_url` ou '`/teachers'`